CMakeLists.txt

```diff
@@ -1,122 +1,42 @@
 # Enable Clang auto-linting, defaults to off.
 option(ENABLE_AUTOLINT "Enable automatic linting and warning with clang-tidy. " OFF)
 # Enable static stack analyzer, defaults to off.
 option(ENABLE_STACK_ANALYSIS "Enable stack analysis. Requires gcc." OFF)
 
-################################################################################
-# Source Code Configuration
-################################################################################
-
-################################################################################
-# Third party.
-################################################################################
+project(libsettings LANGUAGES C CXX)
 
 ################################################################################
-# Compilation Settings
+# Dependency Paths
+# If you wish to use versions other than what's bundled with libsettings
+# set the variables to the path to the root of the different repo version
 ################################################################################
-# Build-type dependent flags.
-set(CMAKE_C_STANDARD 99)
-set(CMAKE_C_FLAGS_RELEASE "-O2")
-set(CMAKE_C_FLAGS_MINSIZEREL "-Os -s")
-set(CMAKE_C_FLAGS_RELWITHDEBINFO "-O2 -g")
-set(CMAKE_CXX_FLAGS_RELEASE "-O2")
-set(CMAKE_CXX_FLAGS_MINSIZEREL "-Os -s")
-set(CMAKE_CXX_FLAGS_RELWITHDEBINFO "-O2 -g")
-set(CMAKE_CXX_STANDARD 11)
-
-# Setup flags for Code Coverage build mode
-set(CMAKE_C_FLAGS_COVERAGE "${CMAKE_C_FLAGS_DEBUG} --coverage" CACHE STRING
-    "Flags used by the C compiler for building with code coverage."
-    FORCE)
-set(CMAKE_EXE_LINKER_FLAGS_COVERAGE
-    "${CMAKE_EXE_LINKER_FLAGS_DEBUG} --coverage" CACHE STRING
-    "Flags used for linking binaries with code coverage."
-    FORCE)
-set(CMAKE_SHARED_LINKER_FLAGS_COVERAGE
-    "${CMAKE_SHARED_LINKER_FLAGS_DEBUG} --coverage" CACHE STRING
-    "Flags used by the shared libraries linker during builds with code coverage."
-    FORCE)
-mark_as_advanced(
-    CMAKE_CXX_FLAGS_COVERAGE
-    CMAKE_C_FLAGS_COVERAGE
-    CMAKE_EXE_LINKER_FLAGS_COVERAGE
-    CMAKE_SHARED_LINKER_FLAGS_COVERAGE)
-
-# Set project version using Git tag and hash.
-execute_process(
-  COMMAND git describe --dirty --tags --always
-  WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}
-  RESULT_VARIABLE GIT_VERSION_FOUND
-  ERROR_QUIET
-  OUTPUT_VARIABLE GIT_VERSION
-  OUTPUT_STRIP_TRAILING_WHITESPACE
-)
-
-if (GIT_VERSION_FOUND)
-  set(VERSION "unknown")
-else (GIT_VERSION_FOUND)
-  set(VERSION ${GIT_VERSION})
-endif (GIT_VERSION_FOUND)
-
-# Some compiler options used globally
-if (NOT MSVC)
-  set(CMAKE_C_FLAGS
-      "-Wall -Wextra -Wno-strict-prototypes -Werror -fno-unwind-tables \
-       -fno-asynchronous-unwind-tables -Wimplicit -Wshadow -Wswitch-default \
-       -Wswitch-enum -Wundef -Wuninitialized -Wcast-align -Wformat=2 \
-       -Wimplicit-function-declaration -Wredundant-decls -Wformat-security \
-       -ggdb ${CMAKE_C_FLAGS}")
-endif (NOT MSVC)
+if (NOT LIBSBP_PATH)
```

Comment:
I would have preferred to simply change to using `add_subdirectory()` and clearly and cleanly make `libsettings` depend on `libsbp` and `libswiftnav`. Unfortunately several of our projects try and define clang formatting targets with the same generic name (i.e. `clang-format-all`). This method is an ok middle ground for the time being, tough I think we should eventually re-examine our use of the clang format targets.